syntax = "proto3";

package SystemPrompt;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service SystemPromptController {
    rpc List(SystemPromptListRequest) returns (stream SystemPrompt) {}
    rpc Create(SystemPrompt) returns (SystemPrompt) {}
    rpc Retrieve(SystemPromptRetrieveRequest) returns (SystemPrompt) {}
    rpc Update(SystemPrompt) returns (SystemPrompt) {}
    rpc Destroy(SystemPrompt) returns (google.protobuf.Empty) {}
}

message SystemPrompt {
    string id = 1;
    google.protobuf.Timestamp created_at = 2;
    google.protobuf.Timestamp updated_at = 3;
    string prompt_name = 4;
    string prompt_content = 5;
}

message SystemPromptListRequest {
}

message SystemPromptRetrieveRequest {
    string id = 1;
}
